### PR TITLE
Revert "Revert "some changes to support fine-tuning on Intel GPU (#88…

### PR DESCRIPTION
**Pull request by harborn**
…)" (#95)"

This reverts commit 63464edc33680edf9327605e93b638540ca9f2b2.

### CODE REVIEW

**Review by xwu99 — APPROVED** [review]

**Review by carsonwang — COMMENTED** [review]
Sorry for late review. I'v added a few more comments. Can you please check them and submit a followup PR?

File: common/trainer/default_trainer.py [hunk 1]
```diff
@@ -170,12 +168,17 @@ def train(self):
...
                         self.lr_scheduler.step()
                     self.optimizer.zero_grad()
-                    if step % log_step == 0:
+
+                    if step % logging_steps == 0:
+                        loss = loss.item()
+                        ppl = math.exp(loss)
                         logger.info(
-                            f"train epoch:[{idx}/{num_train_epochs}]\tstep:[{step}/{total_steps}]\tloss:{loss:.6f}\tppl:{math.exp(loss):.6f}\ttime:{time.time()-start:.6f}"
+                            f"train epoch:[{idx}/{num_train_epochs}]\tstep:[{step}/{total_steps}]\tloss:{loss:.6f}\tppl:{ppl:.6f}\ttime:{time.time()-start:.6f}"
```
[the remark this answers is not in the document]
Comment (harborn) [reply common/trainer/default_trainer.py:176]: OK, will update in other PR.

File: finetune/finetune.py [hunk 2]
```diff
@@ -62,6 +63,14 @@ def get_accelerate_environment_variable(mode: str, config: Union[Dict[str, Any],
     return mode_env_vars[mode]
 
 
+def convert_dtype(dtype: str) -> torch.dtype:
+    supported_dtypes = {"fp16": torch.float16, "bf16": torch.bfloat16, "fp32": torch.float32}
+    if dtype in supported_dtypes:
+        return supported_dtypes[dtype]
+    else:
+        raise ValueError(f"only supported torch.dtype list [{supported_dtypes.keys()}]")
```
[the remark this answers is not in the document]
Comment (harborn) [reply finetune/finetune.py:71]: yes, will update in other PR.

File: finetune/finetune_config.py [hunk 3]
```diff
@@ -54,6 +56,8 @@ class Training(BaseModel):
     resources_per_worker: RayResourceConfig
     accelerate_mode: str
     mixed_precision: str = "no"
+    gradient_accumulation_steps: int
```
[the remark this answers is not in the document]
Comment (harborn) [reply finetune/finetune_config.py:59]: yes, will update in other PR.

File: finetune/finetune.yaml [hunk 4]
```diff
@@ -28,3 +30,5 @@ Training:
   resources_per_worker:
     CPU: 32
   accelerate_mode: CPU_DDP
+  gradient_accumulation_steps: 2
```
[the remark this answers is not in the document]
Comment (harborn) [reply finetune/finetune.yaml:33]: yes, will update in other PR.

File: finetune/finetune.py [hunk 5]
```diff
@@ -62,6 +63,14 @@ def get_accelerate_environment_variable(mode: str, config: Union[Dict[str, Any],
     return mode_env_vars[mode]
 
 
+def convert_dtype(dtype: str) -> torch.dtype:
+    supported_dtypes = {"fp16": torch.float16, "bf16": torch.bfloat16, "fp32": torch.float32}
```
[the remark this answers is not in the document]
Comment (harborn) [reply finetune/finetune.py:67]: updated.

**Pull request merged**